### ISSUE TITLE
Warnings for non-numeric return values

### ISSUE THREAD

**Issue by eserte** [miyagawa/CGI-Compile#18]
If the last expression in the compiled cgi is a non-numeric value, then there are new warnings not seen in older versions of CGI::Compile. Consider the following cgi script, saved as test.cgi:
```
#!/usr/bin/perl -w
use CGI qw(:standard);
print header, start_html, 'hello', end_html;
undef;
```
This yields the following warning:
```
$ perl5.22.1 -MCGI::Compile -e '$sub = CGI::Compile->compile("test.cgi"); $sub->()' > /dev/null 
Use of uninitialized value $rv in sprintf at /home/e/eserte/trash/test.cgi line 8.
```
If `undef` in the cgi script was replaced with a non-numeric string, say `"bla"`, then the warning looks like this:
```
Argument "bla" isn't numeric in sprintf at /home/e/eserte/trash/test.cgi line 8.
```
Probably the easiest solution is to put the right side of CGI/Compile.pm line 100 into a `no warnings` block.

**Comment by miyagawa** [comment]
Thanks for the report. Can you open a pull request with possibly a failing test?

**Comment by rkitover** [comment]
Fixed in master, just so you know it's the -w in the CGI's shebang that triggers this.

**Comment by rkitover** [comment]
@miyagawa 
I improved the test and suppressed another warning in that piece of code.

Should be ready for release.

**Comment by miyagawa** [comment]
Thanks, shipped as 0.21.
